compiler/src/iree/compiler/Codegen/VMVX/VMVXLinkExecutables.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "iree/compiler/Codegen/Utils/LinkingUtils.h"
+#include "iree/compiler/Dialect/VM/IR/VMOps.h"
+#include "iree/compiler/Utils/ModuleUtils.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+
+struct VMVXLinkExecutablesPass
+    : public VMVXLinkExecutablesBase<VMVXLinkExecutablesPass> {
+  VMVXLinkExecutablesPass() = default;
+  void runOnOperation() override {
+    auto moduleOp = getOperation();
+    auto moduleBuilder = OpBuilder::atBlockBegin(moduleOp.getBody());
+
+    auto sourceExecutableOps =
+        llvm::to_vector<8>(moduleOp.getOps<IREE::HAL::ExecutableOp>());
+    if (sourceExecutableOps.size() <= 1) return;
+
+    // Guess a module name, if needed, to make the output files readable.
+    auto moduleName = guessModuleName(moduleOp, "llvm_module");
```

Comment:
Should vmvx use `vmvx_module`?
```suggestion
    // Guess a module name, if needed, to make the output files readable.
    auto moduleName = guessModuleName(moduleOp, "vmvx_module");
```